### PR TITLE
Adding Info class

### PR DESCRIPTION
**Pull request by bekozi**
- Replaced Attribute class backend with Info Fortran API
- Removed Attribute linking from ESMF objects
- Moved Attribute documentation

### CODE REVIEW

File: src/Infrastructure/Array/interface/ESMCI_Array_F.C [hunk 1]
```diff
@@ -750,7 +750,7 @@ extern "C" {
     // return successfully
     if (rc!=NULL) *rc = ESMF_SUCCESS;
   }
-  
+
```
Comment (theurich) [inline src/Infrastructure/Array/interface/ESMCI_Array_F.C:753]: Exclusive white space change.

File: src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90 [hunk 2]
```diff
@@ -52,6 +52,7 @@ module ESMF_ArrayGetMod
   use ESMF_DistGridMod
   use ESMF_RHandleMod
   use ESMF_F90InterfaceMod  ! ESMF Fortran-C++ interface helper
+  use ESMF_InfoMod
```
Comment (theurich) [inline src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90:55]: Unclear why this line is needed here.

File: src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90 [hunk 3]
```diff
@@ -466,7 +467,7 @@ type(ESMF_KeywordEnforcer), optional:: keywordEnforcer ! must use keywords below
       totalLWidthArg, totalUWidthArg, localrc)
     if (ESMF_LogFoundError(localrc, ESMF_ERR_PASSTHRU, &
       ESMF_CONTEXT, rcToReturn=rc)) return
-
```
Comment (theurich) [inline src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90:469]: Probably exclusive white space change (assuming the other line changed in this file is not needed).

File: src/Infrastructure/Array/tests/makefile [hunk 4]
```diff
@@ -7,6 +7,7 @@ run_uni:  run_unit_tests_uni
 LOCDIR	  = src/Infrastructure/Array/tests
 
 .NOTPARALLEL:
+#TESTS_BUILD = $(ESMF_TESTDIR)/ESMF_ArrayInfoUTest
```
Comment (theurich) [inline src/Infrastructure/Array/tests/makefile:10]: There is no ESMF_ArrayInfoUTest, and this looks like something left over from development that should not come in.

File: src/Infrastructure/Array/tests/makefile [hunk 5]
```diff
@@ -18,7 +19,7 @@ TESTS_BUILD   = $(ESMF_TESTDIR)/ESMF_ArrayCreateGetUTest \
                 $(ESMF_TESTDIR)/ESMF_ArrayRedistUTest \
                 $(ESMF_TESTDIR)/ESMF_ArrayRedistPerfUTest \
                 $(ESMF_TESTDIR)/ESMF_ArrayHaloUTest \
-                $(ESMF_TESTDIR)/ESMC_ArrayUTest
```
Comment (theurich) [inline src/Infrastructure/Array/tests/makefile:21]: Probably exclusive white space change.

File: src/Infrastructure/Array/tests/makefile [hunk 6]
```diff
@@ -31,7 +32,7 @@ TESTS_RUN     = RUN_ESMF_ArrayCreateGetUTest \
                 RUN_ESMF_ArrayRedistUTest \
                 RUN_ESMF_ArrayRedistPerfUTest \
                 RUN_ESMF_ArrayHaloUTest \
-                RUN_ESMC_ArrayUTest
```
Comment (theurich) [inline src/Infrastructure/Array/tests/makefile:34]: Probably exclusive white space change.

File: src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90 (hunk 2 above)
Comment (bekozi) [reply src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90]: Nice catch. I forgot to remove it when there was some thought of adding info gets to all the object interfaces. I pushed a fix.

File: src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90 (hunk 3 above)
Comment (bekozi) [reply src/Infrastructure/Array/interface/ESMF_ArrayGet.cppF90:469]: Also fixed.

File: src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90 [hunk 7]
```diff
@@ -3731,7 +3731,7 @@ subroutine ESMF_DistGridGetPLocalDe(distgrid, localDe, keywordEnforcer, &
     
     ! Check init status of arguments
     ESMF_INIT_CHECK_DEEP(ESMF_DistGridGetInit, distgrid, rc)
-
```
Comment (theurich) [inline src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90:3734]: Exclusive white space change.

File: src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90 [hunk 8]
```diff
@@ -3767,7 +3767,7 @@ subroutine ESMF_DistGridGetPLocalDe(distgrid, localDe, keywordEnforcer, &
         deallocate(localDeToDeMap, deToTileMap)
       endif
     endif
-
```
Comment (theurich) [inline src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90:3770]: Exclusive white space change.

File: src/Infrastructure/Array/tests/makefile (hunk 5 above)
Comment (bekozi) [reply src/Infrastructure/Array/tests/makefile:21]: Is there a git configuration to fix this?

File: src/Superstructure/AttributeAPI/makefile [hunk 9]
```diff
@@ -14,7 +14,7 @@ include ${ESMF_DIR}/makefile
 #  directly below this directory, and have either library,
 #  example/test code, or documents which need to be generated.
 #DIRS	  = interface doc examples
```
Comment (theurich) [inline src/Superstructure/AttributeAPI/makefile:16]: Good time to remove this old commented out line.

File: src/addon/NUOPC/src/NUOPC_Base.F90 [hunk 10]
```diff
@@ -3725,7 +3725,6 @@ subroutine NUOPC_RealizeTransfer(state, fieldName, typekind, gridToFieldMap, &
     logical                         :: l_realizeOnlyNotComplete
     logical                         :: isComplete, isConnected, isSharedField
     type(ESMF_FieldStatus_Flag)     :: fieldStatus
-
```
Comment (theurich) [inline src/addon/NUOPC/src/NUOPC_Base.F90:3728]: Exclusive white space change.

File: src/addon/NUOPC/tests/ESMF_NUOPC_UTest.F90 [hunk 11]
```diff
@@ -290,7 +290,7 @@ program ESMF_NUOPC_UTest
   call ESMF_GridCompInitialize(gridComp, userRc=urc, rc=rc)
   if (rc /= ESMF_SUCCESS) call ESMF_Finalize(endflag=ESMF_END_ABORT)
   if (urc /= ESMF_SUCCESS) call ESMF_Finalize(endflag=ESMF_END_ABORT)
-
```
Comment (theurich) [inline src/addon/NUOPC/tests/ESMF_NUOPC_UTest.F90:293]: Exclusive white space change.

File: src/Infrastructure/Array/tests/makefile (hunk 5 above)
Comment (bekozi) [reply src/Infrastructure/Array/tests/makefile:21]: From what I can tell just best to revert...nvm.

File: src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90 [hunk 12]
```diff
@@ -63,7 +63,7 @@ module ESMF_DistGridMod
   sequence
 #endif
   private
-    type(ESMF_Pointer) :: this
+    type(ESMF_Pointer) :: thisq
```
Comment (theurich) [inline src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90:66]: Looks like an accidental 'q' made it to the end of this line.

Comment (bekozi) [reply src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90]: Wondering how that didn't cause more problems...

Comment (theurich) [reply src/Infrastructure/DistGrid/interface/ESMF_DistGrid.F90]: It does if you tried to build DistGrid after the 'q' came in. But now its fixed again.

**Comment by bekozi** [comment]
Thanks for the review! I'll just squash and merge here on the PR.

**Comment by theurich** [comment]
Sounds good.

**Pull request merged**